### PR TITLE
Feature/page filtering

### PR DESCRIPTION
**Pull request by laurejt**
Completed initial implementation of page filtering (including unit tests). I'm especially looking for feedback on how I've selected to have this filtering option interact with the other filtering options.

### CODE REVIEW

File: src/corppa/utils/filter.py [hunk 1]
```diff
@@ -209,9 +244,16 @@ def main():
...
-        help="filename with list of source ids, one per line",
+        help="File containing a list of source ids (one per line) to filter to",
+        type=pathlib.Path,
         required=False,
     )
+    filter_args.add_argument(
+        "--pgfile",
+        help="CSV file containing the list of pages to filter to. File must be headerless with "
+        + "fields (1) source id and (2) digital page number.",
+        type=pathlib.Path,
```
Comment (rlskoeser) [inline src/corppa/utils/filter.py:255]: why headerless?

**Review by rlskoeser — APPROVED** [review]
These updates look good. The refactor to use pathlib objects with argparse is a big improvement, and I like your decision about how to implement the CSV filtering, with just the specific fields we care about.  The unit tests also look great.

Please add example usage at the top of the file explaining the new mode and documenting exactly how the CSV should be structured. (I was confused about the "headerless" note; my preference would be to use defined headers and complain if they are not present). You should also add a brief not explaining how this interacts with the other filters.

File: src/corppa/utils/filter.py [hunk 2]
```diff
@@ -235,25 +277,31 @@ def main():
 
     # at least one filter must be specified
     # check that one of idfile, include, or exclude is specified
-    if not any([args.idfile, args.include, args.exclude]):
+    if not any([args.idfile, args.pgfile, args.include, args.exclude]):
```
[the remark this answers is not in the document]
Comment (rlskoeser) [reply src/corppa/utils/filter.py:280]: I was looking at argparse docs for a different reason - there's a [mutually exclusive group](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_mutually_exclusive_group) option and you can make it required - but I don't think we want these to be mutually exclusive. The argument group is really only for display purposes and can't be set to required.

File: src/corppa/utils/filter.py (hunk 1 above)
Comment (laurejt) [reply src/corppa/utils/filter.py:255]: Not a great reason, I didn't want to decided on a standardized header.

File: src/corppa/utils/filter.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (laurejt) [reply src/corppa/utils/filter.py:280]: Good catch, I missed this one

File: src/corppa/utils/filter.py [hunk 3]
```diff
@@ -33,20 +34,23 @@
 
 
 def filter_pages(
-    input_filename: str,
+    input_filename: pathlib.Path,
     source_ids: list[str] | None = None,
-    disable_progress: bool = False,
+    page_index: dict | None = None,
```
[the remark this answers is not in the document]
Comment (laurejt) [reply src/corppa/utils/filter.py:39]: It's a mapping from volumes to select pages, if that gives any additional inspiration for naming

File: src/corppa/utils/filter.py (hunk 1 above)
Comment (laurejt) [reply src/corppa/utils/filter.py:255]: That's a good point, I'll add it in and document.

**Review by rlskoeser — APPROVED** [review]
Revisions look great! Thanks for adding all the documentation and cleaning up the work id / source id confusion.

I think this is a good start on documentation, how about we merge this and then I can expand in a new branch?

**Pull request merged**